src/Application/Kernel.php

```diff
@@ -8,39 +8,38 @@
 use RecursiveIteratorIterator;
 use ReflectionClass;
 use Tempest\AppConfig;
-use Tempest\Application\Exceptions\KernelException;
 use Tempest\Container\Container;
 use Tempest\Container\GenericContainer;
 use Tempest\Database\PDOInitializer;
-use Tempest\Discovery\DiscoveryLocation;
 use Tempest\Http\RequestInitializer;
 use Tempest\Http\RouteBindingInitializer;
 use Tempest\Http\ServerInitializer;
-use function Tempest\path;
 use Throwable;
 
 final readonly class Kernel
 {
     public function __construct(
-        private string $root,
+        public string $root,
         private AppConfig $appConfig,
+        private array $bootstraps = [],
     ) {
     }
 
     public function init(): Container
     {
-        $container = $this->initContainer();
-
-        $this->initDiscoveryLocations();
+        $container = $this->createContainer();
+        $container->config($this->appConfig);
 
-        $this->initConfig($container);
+        foreach ($this->bootstraps as $bootstrap) {
+            $container->get($bootstrap)->boot();
+        }
 
         $this->initDiscovery($container);
```

Comment:
@brendt Given the extraction of the Discovery/Config code, this straggler `initDisovery()` feels out of place, but what should i do with it?

Honestly, i'm starting to question whether i was on the right track with this proposal at all. Seems to raise concerns and some clumsiness. Any suggestions are welcome, but i'm also okay with abandoning this for some other approach.